### PR TITLE
Update `rollup_vk` Based on Features in `datatool`

### PR DESCRIPTION
**Pull request by prajwolrg**
## Description

This PR updates the rollup_vk implementation in `datatool` to select the verifying key (vk) based on the enabled feature. The supported features are:
- `risc0`: Uses the Risc0 verifying key.
- `sp1`: Uses the SP1 verifying key.

If neither feature is enabled, it defaults to the native.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation update
- [X] Refactor
- [ ] New or updated tests
- [X] Dependency Update

## Checklist

- [X] I have performed a self-review of my code.
- [X] I have commented my code where necessary.
- [X] I have updated the documentation if needed.
- [X] My changes do not introduce new warnings.
- [X] I have added tests that prove my changes are effective or that my feature works.
- [X] New and existing tests pass with my changes.

### CODE REVIEW

File: crates/consensus-logic/src/l1_handler.rs [hunk 1]
```diff
@@ -187,6 +187,7 @@ pub fn verify_proof(checkpoint: &BatchCheckpoint, rollup_params: &RollupParams)
         RollupVerifyingKey::SP1VerifyingKey(vk) => {
             strata_sp1_adapter::verify_groth16(proof, vk.as_ref(), &public_params_raw)
         }
+        RollupVerifyingKey::NativeVerifyingKey(_) => Ok(()),
```
Comment (MdTeach) [inline crates/consensus-logic/src/l1_handler.rs:190]: For now, should we add a comment, considering that in native mode we perform public parameter assertions?

File: bin/datatool/Cargo.toml [hunk 2]
```diff
@@ -10,15 +10,22 @@ path = "src/main.rs"
...
 strata-primitives.workspace = true
-strata-sp1-guest-builder = { path = "../../provers/sp1" }
+strata-risc0-guest-builder = { path = "../../provers/risc0", optional = true }
+strata-sp1-guest-builder = { path = "../../provers/sp1", optional = true }
 
 anyhow.workspace = true
 argh.workspace = true
 bech32 = "0.11.0"
 bitcoin = { workspace = true, features = ["std"] }
+bytemuck = "1.21.0"
```
Comment (MdTeach) [inline bin/datatool/Cargo.toml:20]: since bytemuck is only used with risc0, should we make it optional and feature-gated with risc0?

**Review by delbonis — CHANGES_REQUESTED** [review]
Looks good overall.

File: functional-tests/rollup_params_cfg.py [hunk 3]
```diff
@@ -17,10 +19,21 @@ def get_operators_pubkeys(self) -> list[str]:
...
+class Sp1RollupVk(BaseModel):
     sp1: str
 
 
+class Risc0RollupVk(BaseModel):
+    risc0: str
+
+
+class NativeRollupVk(BaseModel):
+    native: str
```
[the remark this answers is not in the document]
Comment (prajwolrg) [reply functional-tests/rollup_params_cfg.py]: I created a new `StrBuf32` types. Maybe this makes more sense.

File: bin/datatool/src/util.rs [hunk 4]
```diff
@@ -66,6 +65,55 @@ pub(super) fn exec_subc(cmd: Subcommand, ctx: &mut CmdContext) -> anyhow::Result
...
+    }
+
+    // Panic if both `sp1` and `risc0` feature are enabled
+    #[cfg(all(feature = "risc0", feature = "sp1"))]
+    {
+        panic!(
+            "Conflicting ZKVM features: both 'sp1' and 'risc0' are enabled. \
+             Please disable one of them, as only a single ZKVM can be supported at a time."
+        )
+    }
```
[the remark this answers is not in the document]
Comment (storopoli) [reply bin/datatool/src/util.rs:108]: What about this?

```rs
#[cfg_attr(
    all(feature = "risc0", feature = "sp1"),
    compile_error!("Conflicting ZKVM features: both 'sp1' and 'risc0' are enabled. Please disable one of them, as only a single ZKVM can be supported at a time.")
)]
```

Then trying to `allow` the lint specifically for this line of code.

**Review by storopoli — APPROVED** [review]
ACK d6e90828f0d57acbb9c944fd125a570ac70989b9

File: bin/datatool/src/util.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (prajwolrg) [reply bin/datatool/src/util.rs:108]: I could not get it to work with `#cfg_attr`.
I got the error:     
```
compile_error!
             ^ expected one of `(`, `,`, `::`, `=`, `[`, or `{`
```

[the remark this answers is not in the document]
Comment (delbonis) [reply bin/datatool/src/util.rs:108]: Alright it's fine, looks like it's more trouble than it's worth.

**Review by delbonis — APPROVED** [review]

**Pull request merged**